### PR TITLE
Implement language translations for handling multi-language plugin

### LINKED ISSUE #85: Handle multi languages plugin
## Problem

For now the plugin only works in English as the text has been hard-coded inside the different components.

## Solution

Strapi provided a very good way to handle multiple languages. 
It has been tried in the header of the MeiliSearch plugin `meilisearch/admin/src/containers/HomePage/index.js`

```js
<Header
            title={{
              label: formatMessage({ id: getTrad('plugin.name') })
            }}
            content={formatMessage({ id: getTrad('header.description') })}
          />
```

The text is imported from `meilisearch/admin/src/translations`. 

We should change all `hard-coded` text with the same feature.

### Example

```
'Indexed In MeiliSearch'
```

should become
```
// collections.js
formatMessage({ id: getTrad('collection.indexed') })
```

And the following should be added in at least `en`
```js
//meilisearch/admin/src/translations
"collection.indexed": "Indexed in MeiliSearch"
```

## Whats next

Once this is done, it will be very easy to add new languages to the plugin

### PR DESCRIPTION
**Pull request by mimartinez**
# Pull Request

## Related issue
Fixes #85

## What does this PR do?
- Implement language translations for handling multi-language plugin
- Added translation files `en.json` and `es.json`

## PR checklist
Please check if your PR fulfills the following requirements:
- [x] Does this PR fix an existing issue, or have you listed the changes applied in the PR description (and why they are needed)?
- [x] Have you read the contributing guidelines?
- [x] Have you made sure that the title is accurate and descriptive of the changes?

Thank you so much for contributing to Meilisearch!

### CODE REVIEW

**Review by brunoocasali — CHANGES_REQUESTED** [review]
Heey @mimartinez thanks a lot for this PR!

I put some comments, lets discuss them, after that I will merge it :)

Thanks a lot!

File: admin/src/containers/Settings/Credentials.js [hunk 1]
```diff
@@ -1,31 +1,52 @@
...
         <TextInput
-          placeholder="Host"
-          label="Meilisearch Host"
+          placeholder={i18n(
+            'plugin.tab.settings.input.host.placeholder',
+            'Host',
+          )}
+          label={i18n(
+            'plugin.tab.settings.input.host.label',
+            'Meilisearch Host',
```
[the remark this answers is not in the document]
Comment (mimartinez) [reply admin/src/containers/Settings/Credentials.js:23]: It was already like that, so I didn't touch it 😉. But you're right, we changed it 👍

File: admin/src/utils/getTrad.js [hunk 2]
```diff
@@ -0,0 +1,5 @@
+import pluginId from '../pluginId'
+
+const getTrad = id => `${pluginId}.${id}`
+
+export default getTrad
```
[the remark this answers is not in the document]
Comment (mimartinez) [reply admin/src/utils/getTrad.js:5]: It is also used in the `useI18n.js` hook. But you can remove it and do the same in the hook which is what is called everywhere 👌

File: admin/src/index.js [hunk 3]
```diff
@@ -32,4 +35,30 @@ export default {
...
+          })
+          .catch(() => {
+            return {
+              data: {},
+              locale,
+            }
+          })
+      }),
+    )
+    return Promise.resolve(importedTrads)
```
[the remark this answers is not in the document]
Comment (mimartinez) [reply admin/src/index.js:62]: This is the example they have in Strapi Doc for plugin development https://docs.strapi.io/dev-docs/api/plugins/admin-panel-api#registertrads

[the remark this answers is not in the document]
Comment (mimartinez) [reply admin/src/index.js:62]: It is true that the second `Promise.all` is not necessary. I have left it as the example in Strapi's documentation.

**Comment by mimartinez** [comment]
Hi @brunoocasali,
Is there a problem with this PR, can I help? 
Regards

**Comment by curquiza** [comment]
Hello @mimartinez 

@brunoocasali is still on Holiday and will review when he'll come back

**Review by brunoocasali — APPROVED** [review]
Amazing work here @mimartinez 🎉 

bors merge

**Pull request merged**